ReactNativeClient/lib/onedrive-api.js

```diff
@@ -130,6 +130,58 @@ class OneDriveApi {
 		}
 	}
 
+
+	async uploadBigFile(url, options) {
+		const response = await shim.fetch(url, {
+			method: 'POST',
+			headers: {
+				'Authorization': options.headers.Authorization,
+				'Content-Type': 'application/json',
+			},
+		});
+		if (!response.ok) {
+			return response;
+		} else {
+			const data = await response.json();
+			const uploadUrl = data.uploadUrl;
+			// uploading file in 7.5 MiB-Fragments (except the last one) because this is the mean of 5 and 10 Mib which are the recommended lower and upper limits.
+			// https://docs.microsoft.com/de-de/onedrive/developer/rest-api/api/driveitem_createuploadsession?view=odsp-graph-online#best-practices
+			const chunkSize = 7864320;
+			const numberOfChunks = Math.ceil(options.fileSize / chunkSize);
+			const handle = await shim.fsDriver().open(options.path, 'r');
+			for (let i = 0; i < numberOfChunks; i++) {
+				const startByte = i * chunkSize;
+				let endByte = null;
+				let contentLength = null;
+				if (i == numberOfChunks - 1) {
+					// Last fragment. It is not ensured that the last fragment is a multiple of 327,680 bytes as recommanded in the api doc. The reasons is that the docs are out of day for this purpose: https://github.com/OneDrive/onedrive-api-docs/issues/1200#issuecomment-597281253
+					endByte = options.fileSize - 1;
+					contentLength = options.fileSize - ((numberOfChunks - 1) * chunkSize);
+				} else {
+					endByte = (i + 1) * chunkSize - 1;
+					contentLength = chunkSize;
+				}
+				this.logger().info(`${options.path}: Uploading File Fragment ${(startByte / 1048576).toFixed(2)} - ${(endByte / 1048576).toFixed(2)} from ${(options.fileSize / 1048576).toFixed(2)} Mbit ...`);
+				const headers = {
+					'Content-Length': contentLength,
+					'Content-Range': `bytes ${startByte}-${endByte}/${options.fileSize}`,
+					'Content-Type': 'application/octet-stream; charset=utf-8',
+				};
+
+				const response = await shim.uploadChunk(uploadUrl, handle, { contentLength: contentLength, method: 'PUT', headers: headers });
+
+				if (response.ok) {
+					this.logger().info(`${options.path}: ${(endByte / 1048576).toFixed(2)} from ${(options.fileSize / 1048576).toFixed(2)} Mbit have been uploaded`);
```

Comment:
I don't think we should log this as that would be one statement for each chunk. Or maybe make it `logger().debug` so that it doesn't appear in production.